Gems/Automation/Code/Tests/AutomationTests.cpp

```diff
@@ -0,0 +1,141 @@
+
+            AZ::Debug::TraceMessageBus::Handler::BusConnect();
+        }
+
+        void TearDown() override
+        {
+            AZ::Debug::TraceMessageBus::Handler::BusDisconnect();
+
+            m_automationWarnings.set_capacity(0);
+            m_automationLogs.set_capacity(0);
+
+            AutomationApplicationFixture::TearDown();
+        }
+
+    protected:
+        // AZ::Debug::TraceMessageBus implementation
+        bool OnError(const char* window, const char* message) override
+        {
+            AZ_UNUSED(window);
+            AZ_UNUSED(message);
+            // Do nothing. Errors will cause the test to fail. If we suppress errors,
+            // this function won't get called. We test errors by suppressing them and 
+            // then counting how many errors were suppressed.
+            return false;
+        }
+
+        bool OnWarning(const char* window, const char* message) override
+        {
+            if (azstricmp(window, "Automation") == 0)
+            {
+                m_automationWarnings.push_back(message);
+            }
+            return false;
+        }
+
+        bool OnPrintf(const char* window, const char* message) override
+        {
+            if (azstricmp(window, "Automation") == 0)
+            {
+                m_automationLogs.push_back(message);
+            }
+            return false;
+        }
+
+        AZStd::vector<AZStd::string> m_automationWarnings;
+        AZStd::vector<AZStd::string> m_automationLogs;
+    };
+
+    TEST_F(TrackedAutomationFixture, ScriptCommandLineArgument_UsesPrintMethods_AllOperationsLogged)
+    {
+        const char* scriptPath = "print_test.lua";
+        auto application = CreateApplication(scriptPath);
+
+        // We expect our "Hello World" error message to be printed by the test script.
+        // There should be exactly one error message.
+        AZ_TEST_START_TRACE_SUPPRESSION;
+        application->RunMainLoop();
+        AZ_TEST_STOP_TRACE_SUPPRESSION(1);
+
+        AZStd::string executeScriptLog = AZStd::string::format("Running script '%s'...\n", scriptPath);
+        const char* scriptLog = "Script: Hello World\n";
+
+        ASSERT_EQ(m_automationLogs.size(), 2);
+        EXPECT_STREQ(m_automationLogs[0].c_str(), executeScriptLog.c_str());
+        EXPECT_STREQ(m_automationLogs[1].c_str(), scriptLog);
+
+        ASSERT_EQ(m_automationWarnings.size(), 1);
+        EXPECT_STREQ(m_automationWarnings[0].c_str(), scriptLog);
+    }
+
+    TEST_F(TrackedAutomationFixture, ScriptCommandLineArgument_UsesIdleFramesMethod_AllOperationsLogged)
+    {
+        auto application = CreateApplication("idle_five_frames_test.lua");
+
+        application->RunMainLoop();
+
+        ASSERT_EQ(m_automationLogs.size(), 3);
+        // first log is the "running script ..." line
+        EXPECT_STREQ(m_automationLogs[1].c_str(), "Script: Going to idle for 5 frames\n");
+        EXPECT_STREQ(m_automationLogs[2].c_str(), "Script: Idled for 5 frames\n");
```

Comment:
We have a [GoogleTest printer](https://github.com/o3de/o3de/blob/development/Code/Framework/AzTest/AzTest/Printers.h#L25-L30) for our string classes.
You can just use `EXPECT_EQ`